tests/snappi_tests/bgp/conftest.py

```diff
@@ -1,26 +1,264 @@
+    """Execute a command over SSH and check for failures."""
+    stdin, stdout, stderr = ssh.exec_command(command, get_pty=True)
+
+    if password:
+        stdin.write(f"{password}\n")  # Provide sudo password
+        stdin.flush()
+
+    output, error = stdout.read().decode(), stderr.read().decode()
+
+    if error:
+        logger.error(f"Command failed: {command}\nError: {error}")
+        pytest.fail(f"Command execution failed on DUT: {command}")
+
+    return output
+
+
+def are_critical_containers_running(ssh):
+    """Check if all critical containers (bgp, swss, syncd) are running."""
+    running_containers_output = execute_command(
+        ssh, "docker ps -f 'status=running' --format '{{.Names}}'")
+    running_containers = set(running_containers_output.split())
+
+    return CRITICAL_CONTAINERS.issubset(running_containers)
+
+
+def validate_critical_containers(ssh):
+    """Ensure critical containers are running, waiting up to 120s with 10s intervals."""
+    logger.info("Validating critical containers are running...")
+
+    assert wait_until(120, 10, 0, lambda: are_critical_containers_running(
+        ssh)), "Critical containers did not start within 120 seconds!"
+
+    logger.info("All critical containers are running.")
+
+
+def apply_t1_config_on_dut(device_ip, creds,
+                           remote_tmp_path="/tmp/config_db.json.tmp",
+                           final_path="/etc/sonic/config_db.json"):
+    """Move the file to /etc/sonic, rename it, reload config, and validate containers."""
+    username = creds.get('sonicadmin_user')
+    password = creds.get('sonicadmin_password')
+
+    ssh = ssh_connect(device_ip, username, password)
+    exist_prefix_deny = False
+
+    try:
+        logger.info(f"Applying configuration on T1 DUT: {device_ip}...")
+        execute_command(
+            ssh, f"sudo mv {remote_tmp_path} {final_path}", password)
+        logger.info(f"Moved {remote_tmp_path} to {final_path}")
+
+        exist_prefix_deny = execute_command(
+            ssh, "vtysh -c 'show run | include bgp' | grep UPSTREAM_PREFIX_DENY")
+
+        if not exist_prefix_deny:
+            logger.info(
+                f"Apply filter for T1 community {snappi_community_for_t1_drop[0]}")
+            execute_command(ssh,
+                            "vtysh" + " -c ".join([
+                                "config t",
+                                f"bgp community-list standard UPSTREAM_PREFIX_DENY {snappi_community_for_t1_drop[0]}",
+                                "route-map FROM_TIER2_V4 deny 10",
+                                "match community UPSTREAM_PREFIX_DENY",
+                                "exit",
+                                "route-map FROM_TIER2_V6 deny 10",
+                                "match community UPSTREAM_PREFIX_DENY"
+                                ])
+                            )
+
+        execute_command(ssh, "sudo config reload -y", password)
```

Comment:
wouldnt config_reload clean up above vtysh config?

Reply:
that's right. I have fixed this part here